WordPress/Classes/Services/NotificationSyncMediator.swift

```diff
@@ -312,30 +314,30 @@ private extension NotificationSyncMediator {
     ///     - completion: Callback to be executed on completion
     ///
     func determineUpdatedNotes(with remoteHashes: [RemoteNotification], completion: @escaping (([String]) -> Void)) {
-        let derivedContext = type(of: self).sharedDerivedContext(with: contextManager)
-
-        derivedContext.perform {
-            let remoteIds = remoteHashes.map { $0.notificationId }
-            let predicate = NSPredicate(format: "(notificationId IN %@)", remoteIds)
-            var localHashes = [String: String]()
-
-            for note in derivedContext.allObjects(ofType: Notification.self, matching: predicate) {
-                localHashes[note.notificationId] = note.notificationHash ?? ""
-            }
+        Self.operationQueue.addOperation(AsyncBlockOperation { [contextManager] done in
+            contextManager.performAndSave { context in
+                let remoteIds = remoteHashes.map { $0.notificationId }
+                let predicate = NSPredicate(format: "(notificationId IN %@)", remoteIds)
+                var localHashes = [String: String]()
+
+                for note in context.allObjects(ofType: Notification.self, matching: predicate) {
+                    localHashes[note.notificationId] = note.notificationHash ?? ""
+                }
 
-            let filtered = remoteHashes.filter { remote in
-                let localHash = localHashes[remote.notificationId]
-                return localHash == nil || localHash != remote.notificationHash
-            }
+                let filtered = remoteHashes.filter { remote in
+                    let localHash = localHashes[remote.notificationId]
+                    return localHash == nil || localHash != remote.notificationHash
+                }
 
-            derivedContext.reset()
+                let outdatedIds = filtered.map { $0.notificationId }
```

Comment:
Sort of outside the scope, and mostly a nitpick, but I just noticed that there's no need to store `filtered` because it's used only to computed `outdatedIds`.

```swift
let outdatedIds = remoteHashes.filter { remote in
    let localHash = localHashes[remote.notificationId]
    return localHash == nil || localHash != remote.notificationHash
 }
.map { $0.notificationId)
```